docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,434 @@ sidebar:
+
+```tact
+import "@stdlib/deploy";
+
+/// https://docs.dedust.io/reference/tlb-schemes#message-deposit_liquidity-1
+message(0x40e108d6) JettonDepositLiquidity {
+    // Pool type: 0 for volatile, 1 for stable
+    // Volatile pool is based on the "Constant Product" formula
+    // Stable-swap pool is optimized for assets of near-equal value,
+    // e.g. USDT/USDC, TON/stTON, etc.
+    poolType: Int as uint1;
+
+    // Provided assets
+    asset0: Asset;
+    asset1: Asset;
+
+    // Minimal amount of LP tokens to be received
+    // If there's less liquidity provided, the provisioning will be rejected
+    // Defaults to 0, makes this value ignored
+    minimalLpAmount: Int as coins = 0;
+
+    // Target amount of the first asset
+    targetBalances0: Int as coins;
+
+    // Target amount of the second asset
+    targetBalances1: Int as coins;
+
+    // Custom payload to attack to the transaction if the provisioning is successful
+    // Defaults to `null`, which means no payload
+    fulfillPayload: Cell? = null;
+
+    // Custom payload to attack to the transaction if the provisioning is rejected
+    // Defaults to `null`, which means no payload
+    rejectPayload: Cell? = null;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#message-deposit_liquidity
+message(0xd55e4686) NativeDepositLiquidity {
+    // Unique identifier used to trace transactions across multiple contracts
+    // Defaults to 0, which means we don't mark messages to trace their chains
+    queryId: Int as uint64 = 0;
+
+    // Toncoin amount for the deposit
+    amount: Int as coins;
+
+    // Inlined fields of JettonDepositLiquidity Message without the opcode prefix
+    poolType: Int as uint1;
+    asset0: Asset;
+    asset1: Asset;
+    minimalLpAmount: Int as coins = 0;
+    targetBalances0: Int as coins;
+    targetBalances1: Int as coins;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#asset
+struct Asset {
+    // Specify 0 for native (TON) and omit all following fields
+    // Specify 1 for Jetton and then you must set non-null values for the following fields
+    type: Int as uint4;
+
+    workchain: Int? as uint8 = null;
+    address: Int? as uint256 = null;
+}
+
+const PoolTypeVolatile: Int = 0;
+const PoolTypeStable: Int = 1;
+
+const AssetTypeNative: Int = 0b0000;
+const AssetTypeJetton: Int = 0b0001;
+
+const JettonProvideLpGas: Int = ton("0.5");
+const JettonProvideLpGasFwd: Int = ton("0.4");
+const TonProvideLpGas: Int = ton("0.15");
+
+// This example directly uses the provided `myJettonWalletAddress`
+// In real-world scenarios, it's more reliable to calculate this address on-chain or save it during initialization to prevent any issues
+fun provideLiquidity(myJettonWalletAddress: Address) {
+    let jettonMasterRaw = parseStdAddress(
+        address("kQDkRHlWaibL7Tww48T6xAUFevPflca7i8TIiQTacBmnDOrb")
+        .asSlice()
```

Comment:
It is not JettonMaster, and why it is from testnet? It's a mainner pool address, in testnet-only form

Reply:
Fixed. Why this discussion is not marked "Outdated"?